main.go

```diff
@@ -39,67 +43,69 @@ func setFlagset(env *agent.Env, cmdargs *CommandArgs) {
 
 	defaultFlags.BoolVar(&env.DryRun, "dry-run", false, "Only print, and do not execute, potentially destructive commands")
 	// -version will always override all other args
-	defaultFlags.BoolVar(&cmdargs.DisplayVersion, "version", false, "Display version information")
+	defaultFlags.BoolVar(&displayVersionFlagged, "version", false, "Display version information")
 
 	if !env.Prod {
 		defaultFlags.StringVar(&env.BaseUrl, "url", env.BaseUrl, "Set the endpoint")
 	}
 
+	defaultFlags.Parse(os.Args[argRange:])
+
+	if displayVersionFlagged {
+		*performCmd = PerformDisplayVersion
```

Comment:
It sucks that you have to mutate performCmd from within this function. 

I kind of thing you should add "-version" to the switch statement below. What do you think?